src/matchmaking-threads.ts

```diff
@@ -0,0 +1,144 @@
+import { MatchmakingThread } from '@/models/matchmakingThreads';
+import { User } from '@/models/users';
+import { sendEventLogMessage } from '@/util';
+import { getDB } from './db';
+import { ChannelType, EmbedBuilder } from 'discord.js';
+import type { Client, Message } from 'discord.js';
+
+export async function handleMatchmakingThreadMessage(message: Message): Promise<void> {
+	const matchmakingChannelId = getDB().get('channels.matchmaking');
+	if (!matchmakingChannelId) {
+		console.log('Missing matchmaking channel!');
+		return;
+	}
+
+	if (message.channel.type !== ChannelType.PublicThread || message.channel.parentId !== matchmakingChannelId) {
+		return;
+	}
+
+	await MatchmakingThread.upsert({ thread_id: message.channelId, last_message_sent: message.createdAt });
+}
+
+export async function checkMatchmakingThreads(client: Client): Promise<void> {
+	let matchmakingLockTimeout = parseInt(getDB().get('matchmaking.lock-timeout-seconds') ?? '0');
+	if (!matchmakingLockTimeout) {
+		console.log('Missing matchmaking lock timeout! Defaulting to 1 hour.');
+		matchmakingLockTimeout = 60 * 60;
+	}
+
+	const now = Date.now();
+	const matchmakingThreads = await MatchmakingThread.findAll();
+
+	for (const thread of matchmakingThreads) {
+		let threadChannel;
+		try {
+			threadChannel = await client.channels.fetch(thread.thread_id);
+		} catch (err) {
+			console.log(`Error fetching matchmaking thread channel: ${err}`);
+			threadChannel = null;
+		}
+
+		if (!threadChannel || threadChannel.type !== ChannelType.PublicThread) {
+			// * Thread is either deleted or not a public thread, which should not happen
+			console.log(`Removing deleted or invalid matchmaking thread from database: ${thread.thread_id}`);
+			await MatchmakingThread.destroy({ where: { thread_id: thread.thread_id } });
+			continue;
+		}
+
+		const lastMessageSent = new Date(thread.last_message_sent).getTime();
```

Comment:
Just for clarity this should probably actually note that it's time-related and not message-related. Right now this variable reads as possibly being the actual message that was sent last, not the time that it was sent at